src/elements/Violin.ts

```diff
@@ -70,32 +70,44 @@ export class Violin extends StatsBase<IViolinElementProps, IViolinElementOptions
     } else {
       maxEstimate = props.maxEstimate;
     }
+    const firstCoord = props.coords[0];
+    const lastCoord = props.coords[props.coords.length - 1];
+
+    ctx.beginPath();
     if (this.isVertical()) {
       const { x, width } = props;
       const factor = width / 2 / maxEstimate;
-      ctx.moveTo(x, props.min);
+      ctx.moveTo(x - firstCoord.estimate * factor, firstCoord.v);
       props.coords.forEach((c) => {
         ctx.lineTo(x - c.estimate * factor, c.v);
       });
-      ctx.lineTo(x, props.max);
-      ctx.moveTo(x, props.min);
-      props.coords.forEach((c) => {
-        ctx.lineTo(x + c.estimate * factor, c.v);
-      });
-      ctx.lineTo(x, props.max);
+
+      ctx.lineTo(x - lastCoord.estimate * factor, props.max);
```

Comment:
if the first one starts at firstCoord.v, shouldn't it end at lastCoord.v, too?

Reply:
After review it seems like the firstCoord and lastCoord are redundant and covered by the .forEach and .reverse().forEach.

I have added a new commit with the fix.